Robot Importer (URDF) first version

adamdbrw opened the pull request:
Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

Work towards #102

Remaining work for a develop merge candidate:
1. Add checks for prefab / file existence, handle possible scenarios and errors.
2. Replace dummy cube collider with URDF based entities reflecting the structure (even if empty for the first version)

adamdbrw committed:
Robot Importer (URDF) widget, components. Initial frames

Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

adamdbrw committed:
Prototype of the importer

Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

adamdbrw committed:
reversed unintended prefab changes

Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

piotr-zyskowski-rai commented on Code/Source/URDF/RobotImporter/ROS2RobotImporterEditorSystemComponent.cpp:
```diff
@@ -0,0 +1,62 @@
+        AzToolsFramework::RegisterViewPane<RobotImporterWidget>("Robot Importer", "ROS2", options);
+    }
+
+    ROS2RobotImporterEditorSystemComponent::ROS2RobotImporterEditorSystemComponent() = default;
+
+    ROS2RobotImporterEditorSystemComponent::~ROS2RobotImporterEditorSystemComponent() = default;
```
Wouldn't it be better to default those in the header file?

piotr-zyskowski-rai commented on Code/Source/URDF/RobotImporter/RobotImporterWidget.cpp:
```diff
@@ -0,0 +1,70 @@
+
+//#include <AzToolsFramework/ToolsComponents/EditorLockComponent.h>
+//#include <AzToolsFramework/ToolsComponents/EditorVisibilityComponent.h>
+//#include <AzToolsFramework/ToolsComponents/TransformComponent.h>
```
To be removed?

adamdbrw committed:
Dynamic prefab creation through URDF import
- Creating entity structure reflecting Link hierarchy
- Creating necessary components for entities (such as TransformComponent)
- Creating ROS2FrameComponent for each link with proper frame_id
- Setting Transforms to pose values from urdf joints (proper placement of parts)
- Component activation

Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

adamdbrw committed:
Inital work on mesh component (visuals)

Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

adamdbrw committed:
URDF visual primitives (#118)

* Primitives. Also separated out type conversions

Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

* removed unnecessary explicit casts

Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

adamdbrw committed:
Urdf importer inertial (#120)

* added inertial to EditorRigidBodyComponent

Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

* applied clang format

Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

* applied formatting

Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

adamdbrw commented:
Closing in favor of https://github.com/RobotecAI/o3de-ros2-gem/pull/121
